changes/tasks.py

```diff
@@ -544,3 +554,89 @@ def run(self, change_id, **kwargs):
             return False
 
 validate_implement = ValidateImplement()
+
+
+class PushMomconnectOptoutToJembi(Task):
+    """
+    Sends a momconnect optout change to Jembi.
+    """
+    name = "ndoh_hub.changes.tasks.push_momconnect_optout_to_jembi"
+    l = get_task_logger(__name__)
+    URL = "%s/optout" % settings.JEMBI_BASE_URL
+
+    def get_today(self):
+        return datetime.datetime.today()
+
+    def get_timestamp(self):
+        return self.get_today().strftime("%Y%m%d%H%M%S")
+
+    def get_identity_address(self, identity, address_type='msisdn'):
+        """
+        Returns a single address for the identity for the given address
+        type.
+        """
+        addresses = identity.get(
+            'details', {}).get('addresses', {}).get(address_type, {})
+        address = None
+        for address, details in iteritems(addresses):
+            if details.get('default'):
+                return address
+        return address
+
+    def get_optout_reason(self, reason):
+        """
+        Returns the numerical Jembi optout reason for the given optout reason.
+        """
+        return {
+            'miscarriage': 1,
+            'stillbirth': 2,
+            'babyloss': 3,
+            'not_useful': 4,
+            'other': 5,
+            'unknown': 6,
+            'job_change': 7,
+            'number_owner_change': 8,
+            'not_hiv_pos': 9,
+        }.get(reason)
+
+    def build_jembi_json(self, change):
+        identity = is_client.get_identity(change.registrant_id) or {}
+        address = self.get_identity_address(identity)
+        return {
+            'encdate': self.get_timestamp(),
+            'mha': 1,
+            'swt': 1,
+            'cmsisdn': address,
+            'dmsisdn': address,
```

Comment:
From what I can see, no, the USSD app doesn't allow the health care worker to opt someone out. But I could have missed something.